mbf_abstract_nav/src/abstract_controller_execution.cpp

```diff
@@ -271,179 +323,178 @@ namespace mbf_abstract_nav
     std::vector<geometry_msgs::PoseStamped> plan;
     if (!hasNewPlan())
     {
-      setState(NO_PLAN);
-      moving_ = false;
-      ROS_ERROR("robot navigation moving has no plan!");
-    }
+      boost::chrono::thread_clock::time_point loop_start_time = boost::chrono::thread_clock::now();
 
-    ros::Time last_valid_cmd_time = ros::Time();
-    int retries = 0;
-    int seq = 0;
+      if (cancel_)
+      {
+        if (force_stop_on_cancel_)
+        {
+          publishZeroVelocity(); // command the robot to stop on canceling navigation
+        }
```

Comment:
we publish zero velocity on cancel in `controller_action.cpp` in `CANCELED` state.